### ISSUE TITLE
Windows 10: Toast notifications are not announced if NvDA starts while ShellExperienceHost or SearchUI are possibly suspended

### ISSUE THREAD

**Issue by josephsl** [nvaccess/nvda#5688]
Hi,
I don't know how to describe this, but the following issue can be reproduced consistently and there is a workaround for this:

Under some circumstances, when NVDA starts, toast/Action Center notifications are not announced in Windows 10.

STR:
1. Start NVDA.
2. Perform an action that causes toasts to appear such as inserting a USB flash drive.
- Expected: NvDA announces toast notifications.
- Actual: No announcement.

Specs:
- OS: Windows 10 Version 1511 (10586)
- NVDA version: master, next.13146.
- Reproducible: Yes with a caveat (see below).
- Reported by others: Yes.

Workaround: Opening and closing Start Menu fixes this problem until NVDA is restarted.

Possible cause: Unlike desktop apps, WinRT and universal apps can be suspended, which helps in reducing power consumption but quite undesirable for screen readers. When this happens, a screen reader will not know that the app is "suspended" in the background and will treat it as "not running". In case of NVDA, NVDA will not know that these processes are running until an event is fired from "suspended" processes (in our case, shellexperiencehost.exe/searchui.exe), which explains why the workaround noted above works.

Thanks.

**Comment by jcsteh** [comment]
It's true that these apps can be suspended, but the event fired by the 
Toast should itself be enough. NVDA doesn't care whether an app is 
"running"; it cares about events.

It'd be good if you can check whether this problem occurs with Narrator. 
I suspect it will. If that is the case, UIA (probably UIA Core) is 
simply failing to fire the events for some reason.

You'll also notice that the first time you alt+tab (don't let go of alt) 
after starting NVDA, it doesn't read. That also happens with Narrator. 
Again, no UIA event is fired. We've reported this one to Microsoft in 
July last year, but we've heard nothing back about it yet.

**Comment by josephsl** [comment]
Hi,

Regarding Alt+TAB problem: confirmed.

In regards to toasts: Narrator exhibits the same behavior if:

* You start Narrator and perform an action that should cause toasts to appear.
* You repeat this action. Narrator then announces toast notifications.

Same workaround works for NVDA (I'll change the issue title if possible). Thanks.

**Comment by jcsteh** [comment]
In that case, this is something Microsoft will need to fix. We'll report it to them.

**Comment by leonardder** [comment]
@josephsl and @jcsteh: Any status updates here?

**Comment by jcsteh** [comment]
Nothing to report. As [noted above](https://github.com/nvaccess/nvda/issues/5688#issuecomment-172717184), this will need to be reported to Microsoft. CC @MichaelDCurran.

**Comment by josephsl** [comment]
Hi, I wonder if it is fixed somewhat with Version 1709 or latest WIP (I’m sure it is, but as Jamie pointed out, this is something MS may know better). Thanks.

**Comment by Adriani90** [comment]
@josephsl can you still reproduce this issue? I think there were some changes to toast notifications in 1803 update or 1809 update. @qchristensen did anyone at NV Access notified Microsoft about this? Or should we do it via feedback hub? The problem is that many people cannot specify exact technical details on how it should be implemented to work properly with NVDA. So I think you are in a better position to report this.

**Comment by josephsl** [comment]
Hi, not exactly anymore. Thanks.

**Comment by Adriani90** [comment]
Ok, not exactly means there are some exceptions? :) I will better let it to you to close this issue or not.

**Comment by josephsl** [comment]
Hi,

at least for now, I'm closing it until someone running 19H1 can confirm it is truly no more. Thanks.
